### DISCUSSION TITLE
Feature request in v6 Slurm.

### DISCUSSION THREAD

**Discussion by ek-nag** [GoogleCloudPlatform/cluster-toolkit#2851] [Ideas and feature requests]
Hello,

We are working with a client setting up their production HPC cluster using `cluster-toolkit`. They are lacking some features in the v6 slurm modules.

- [ ] **Custom Topology Configuration:** Allow the use of a custom `topology.conf` template file in the Slurm cluster blueprint.
- [ ] **Custom Slurm Configuration:** Enable the use of a custom `slurm.conf` template file in the Slurm cluster blueprint to add hybrid partitions and other configurations not supported by the cluster toolkit.
- [ ] **Customer-Managed Encryption Keys (CMEK):** Support the use of CMEK with Slurm cluster nodes.
- [ ] **Specify Slurmctld Port:** Provide an option to set the `Slurmctld` port in the blueprint when creating a Slurm cluster.
- [ ] **High Availability Slurmctld Nodes:** Offer an option to have high availability (HA) `Slurmctld` nodes.
- [ ] **Naming Convention:** Allow specifying a naming convention for the Slurm cluster in the blueprint. For example, specific prefix that all nodes will have including controller and login nodes.

We are wondering if there is plan to implement or work on any of these features/options?
Thanks :)

**Comment by cdunbar13** [comment]
Hello, and thanks for your message/request.  I will reach out to the team to discuss this.

**Reply by cboneti** [reply]
on the naming convention, what is the delta between what is provided by [slurm_cluster_name](https://github.com/GoogleCloudPlatform/cluster-toolkit/tree/develop/community/modules/scheduler/schedmd-slurm-gcp-v6-controller#input_slurm_cluster_name) and what is being asked?

**Reply by ek-nag** [reply]
Controller always get "-controller" added to the hostname. https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/main/community/modules/scheduler/schedmd-slurm-gcp-v6-controller/controller.tf#L99. It would great if we could supply name for controller/controllers.

Also, client wants name login nodes differently that compute nodes. For example, login nodes: tllu001 - tllu009, compute: tllc001 - tllc399.
